Snowflake.Data/Core/SFSession.cs

```diff
@@ -92,15 +92,38 @@ internal Uri BuildLoginUrl()
 
         internal SFSession(String connectionString, SecureString password, IRestRequester restRequester)
         {
+
             this.restRequester = restRequester;
             properties = SFSessionProperties.parseConnectionString(connectionString, password);
 
-            ParameterMap = new Dictionary<SFSessionParameter, object>();
-            ParameterMap[SFSessionParameter.CLIENT_VALIDATE_DEFAULT_PARAMETERS] = 
-                Boolean.Parse(properties[SFSessionProperty.VALIDATE_DEFAULT_PARAMETERS]);
+            try
+            {
+                ParameterMap = new Dictionary<SFSessionParameter, object>();
+                ParameterMap[SFSessionParameter.CLIENT_VALIDATE_DEFAULT_PARAMETERS] =
+                    Boolean.Parse(properties[SFSessionProperty.VALIDATE_DEFAULT_PARAMETERS]);
+
+                int timeoutInSec = int.Parse(properties[SFSessionProperty.CONNECTION_TIMEOUT]);
+                int recommencedMinTimeoutSec = BaseRestRequest.DEFAULT_REST_RETRY_MINUTE_TIMEOUT * 60;
+                if (timeoutInSec < recommencedMinTimeoutSec)
+                {
+                    logger.Warn($"Connection timeout provided is less than recommended minimum value of" +
+                        $" {recommencedMinTimeoutSec}");
+                }
+                else if (timeoutInSec < 0)
+                {
+                    logger.Warn($"Connection timeout provided is negative. Timeout will be infinite.");
+                }
+
+                connectionTimeout = timeoutInSec > 0 ? TimeSpan.FromSeconds(timeoutInSec) : Timeout.InfiniteTimeSpan;
 
-            int timeoutInSec = int.Parse(properties[SFSessionProperty.CONNECTION_TIMEOUT]);
-            connectionTimeout = timeoutInSec > 0 ? TimeSpan.FromSeconds(timeoutInSec) : Timeout.InfiniteTimeSpan;
+            }
+            catch (Exception e)
+            {
+                logger.Error(e.Message);

```

Comment:
I don't understand this try-catch block. What kind of error could this block throw?

Reply:
Then I would prefer a smaller try catch block. What do you think?

Reply:
Yes, I think we can have it around the Boolean.Parse and int.Parse. There should not be any NPE or other things that would throw in the code after that.